File: Ch3/remotelib/server/examplelibrary.py
#!/usr/bin/env python

import os
import sys

class ExampleRemoteLibrary:
  def strings_should_be_equal(self, str1, str2):
    print "Comparing '%s' to '%s'" % (str1, str2)
    if str1 != str2:
      raise AssertionError("Given strings are not equal")
    else:
      return "Given strings are equal"


if __name__ == '__main__':
  from robotremoteserver import RobotRemoteServer
  RobotRemoteServer(ExampleRemoteLibrary(), *sys.argv[1:])

File: Ch3/remotelib/server/examplelibrary.rb
#!/usr/bin/env ruby

class ExampleRemoteLibrary

  def strings_should_be_equal(str1, str2)
    puts "Comparing '#{str1}' to '#{str2}'"
    if str1 != str2
      raise RuntimeError, "Given strings are not equal"
    else
      return "Given strings are equal"
    end
  end
end

if __FILE__ == $0
  require "./robotremoteserver"
  RobotRemoteServer.new(ExampleRemoteLibrary.new, *ARGV)
end
